### PR TITLE
Support preload modules in Nanny

### PR DESCRIPTION
**Pull request by mrocklin**
Today we can run preload modules on workers or the scheduler, but not on the Nanny.

Now we can also run preload scripts on the nanny

```
dask-worker ... --preload-nanny my_module
```

### CODE REVIEW

File: distributed/nanny.py [hunk 1]
```diff
@@ -247,6 +263,10 @@ def local_dir(self):
         await self.listen(self._start_address, listen_args=self.listen_args)
         self.ip = get_address_host(self.address)
 
+        await preloading.on_start(
+            self._preload_modules, self, argv=self.preload_argv,
```
Comment (jrbourbeau) [inline distributed/nanny.py:267]:
```suggestion
            self._preload_modules, self, argv=self.preload_nanny_argv,
```

File: distributed/worker.py [hunk 2]
```diff
@@ -452,7 +452,7 @@ def __init__(
         # Target interface on which we contact the scheduler by default
         # TODO: it is unfortunate that we special-case inproc here
         if not host and not interface and not scheduler_addr.startswith("inproc://"):
-            host = get_ip(get_address_host(scheduler_addr))
+            host = get_ip(get_address_host(scheduler_addr.split("://")[-1]))
```
Comment (jrbourbeau) [inline distributed/worker.py:455]: Why was this change needed?

File: distributed/cli/dask_worker.py [hunk 3]
```diff
@@ -216,6 +216,14 @@
...
     "preload_argv", nargs=-1, type=click.UNPROCESSED, callback=validate_preload_argv
 )
+@click.option(
+    "--preload-nanny",
+    type=str,
+    multiple=True,
+    is_eager=True,
+    help="Module that should be loaded by each nanny "
+    'like "foo.bar" or "/path/to/foo.py"',
+)
```
Comment (jrbourbeau) [inline distributed/cli/dask_worker.py:226]: Currently `preload_nanny_argv` can't be specified on the command line. Do you have any thoughts on how we could support a `--preload-nanny-argv` CLI option? To be clear, I suspect this may be difficult with our current `click` setup and don't think the lack of a `--preload-nanny-argv` option should stop us from merging this PR

File: distributed/worker.py (hunk 2 above)
Comment (mrocklin) [reply distributed/worker.py:455]: It's not entirely related.  I was testing this with another workload and ran into a small bug here.

File: distributed/cli/dask_worker.py (hunk 3 above)
Comment (mrocklin) [reply distributed/cli/dask_worker.py:226]: Yeah, I wasn't sure how to pass things through from click.  My sense is that we probably have to point users to the configuration to handle this.  

I've added https://github.com/dask/dask/pull/6077 to help with documentation here.

File: distributed/distributed.yaml [hunk 4]
```diff
@@ -71,6 +71,10 @@ distributed:
       pause: 0.80  # fraction at which we pause worker threads
       terminate: 0.95  # fraction at which we terminate the worker
 
+  nanny:
+    preload: []
```
Comment (martindurant) [inline distributed/distributed.yaml:75]: Most entries have a comment describing them

Comment (mrocklin) [reply distributed/distributed.yaml]: Thanks for flagging this.

I've gone ahead and added brief comments that mostly just point to documentation (I don't think that we can reasonably describe this in one line).  Ben and I are planning to play with larger descriptions for documentation later this week though.
